src/NServiceBus.Core/Transports/Msmq/QueuePermissions.cs

```diff
@@ -43,34 +42,30 @@ public static void CheckQueue(string address)
                 Logger.Warn($"Unable to verify queue at address '{queuePath}'. Make sure the queue exists, and that the address is correct. Processing will still continue.", ex);
             }
         }
-
-        static void WarnIfPublicAccess(MessageQueue queue)
+        
+        static void WarnIfPublicAccess(MessageQueue queue, string userGroupName)
         {
-            MessageQueueAccessRights? everyoneRights, anonymousRights;
+            MessageQueueAccessRights? accessRights;
+            AccessControlEntryType? accessType;
 
             try
             {
-                queue.TryGetPermissions(LocalAnonymousLogonName, out anonymousRights);
-                queue.TryGetPermissions(LocalEveryoneGroupName, out everyoneRights);
+                queue.TryGetPermissions(userGroupName, out accessRights, out accessType);
             }
             catch (SecurityException se)
             {
                 Logger.Warn($"Unable to read permissions for queue [{queue.QueueName}]. Make sure you have administrative access on the target machine", se);
                 return;
             }
 
-            if (anonymousRights.HasValue || everyoneRights.HasValue)
-            {
-                var logMessage = $"Queue [{queue.QueueName}] is running with [{LocalEveryoneGroupName}] and/or [{LocalAnonymousLogonName}] permissions. Consider setting appropriate permissions, if required by the organization. For more information, consult the documentation.";
+            if (accessType == AccessControlEntryType.Deny) return;
 
-                if (Debugger.IsAttached)
-                {
-                    Logger.Info(logMessage);
-                }
-                else
-                {
-                    Logger.Warn(logMessage);
-                }
+            if (accessRights?.HasFlag(MessageQueueAccessRights.GenericRead) == true ||
```

Comment:
@andreasohlund - Yeah, will do. Now that I have made the necessary changes to expose the AclType it should be easy. I meant to make that change :(